### DISCUSSION TITLE
How can i hide too many instances?

### DISCUSSION THREAD

**Discussion by Eikosa** [spyoungtech/ahk#156] [Q&A] [answered]
```
from ahk import AHK, Hotkey

ahk = AHK()


def hotkey_master(hotkey, script):
    hotkey = Hotkey(ahk, hotkey, script) # Create Hotkey
    hotkey.start()  #  Start listening for hotkey


    
hotkey_master("#Up","send, {RAlt Up}{PgUp}")
hotkey_master("#Down","send, {RAlt Down}{PgDn}")
hotkey_master("#Left","send, {RAlt Left}{PgUp}")
hotkey_master("#Right","send, {RAlt Right}{PgDn}")
hotkey_master("!Space","send, {Media_Play_Pause}")

import time

print("running")
while True:
    time.sleep(1)
```

It looks like this: 
![image](https://user-images.githubusercontent.com/20538090/173954466-2dd66953-f73c-42a8-95de-363e7d1be48e.png)

**Comment by spyoungtech** [comment, marked as answer]
You can add the `NoTrayIcon ` directive to hide the tray icon altogether.

```python
from ahk import AHK
from ahk.directives import NoTrayIcon

ahk = AHK(directives=[NoTrayIcon])
```

Also note that you may see multiple icons even if only one instance is running. If you mouse over the icons, the extras will probably disappear.

**Reply by i448** [reply]
Can the ` AHKDaemon ` also use it?
Like this?
``` ahk = AHKDaemon(executable_path=path/to/.exe", directives=[NoTrayIcon]) ```

**Reply by spyoungtech** [reply]
If I recall, `AHKDaemon` doesn't make use of the `directives` keyword, but this should be resolved in the upcoming v1 release (where the daemon strategy becomes the default).

**Comment by spyoungtech** [comment]
V1 is now released and will only create 1 process to handle all command invocations by default (similar to daemon mode in v0) -- directives are also now supported.
